### PR TITLE
Add date and time

### PR DESCRIPTION
**Pull request by sprkweb**
https://github.com/JasonPuglisi/lastfm-now/issues/14

### CODE REVIEW

File: public/javascripts/now.js [hunk 1]
```diff
@@ -559,11 +561,11 @@ function updateMetadata() {
   if (nowPlaying()) {
     document.title = '"' + title + '" by ' + artist;
     if (cookieEnabled('extendedOn')) {
-      toggleDisplay('#userLine', true);
+      toggleDisplay(resources.state.extendedInfo, true);
     }
   } else {
     document.title = 'Last.fm Now';
-    toggleDisplay('#userLine', false);
+    toggleDisplay(resources.state.extendedInfo, false);
```
Comment (JasonPuglisi) [inline public/javascripts/now.js]: I realized making this a part of the extended info will cause it to disappear when nothing is playing. Could you make it a separate hotkey like `t`? Remember to add it to the help menu and create a cookie for it.

File: public/javascripts/now.js [hunk 2]
```diff
@@ -156,7 +158,7 @@ function processKey(event) {
         toggleCookie('extendedOn');
         showE = undefined;
       }
-      toggleDisplay('#userLine', showE);
+      toggleDisplay(resources.state.extendedInfo, showE);
```
Comment (JasonPuglisi) [inline public/javascripts/now.js]: If you add a new hotkey `t`, can you add something like this to the top of the `processKey` function to make sure it doesn't get triggered when doing something like `ctrl`+`t`?

```javascript
// Return if modifier key is held down
if (window.event.ctrlKey || window.event.metaKey)
  return;
```

File: public/javascripts/now.js [hunk 3]
```diff
@@ -670,6 +672,46 @@ function updateWeather(coords, iconMap) {
...
+  updateDatetime();
+}
+
+function updateDatetime() {
+  $('#datetime #date').text(getCurrentDate());
+  $('#datetime #time').text(getCurrentTime());
+
+  setTimeout(function() {
+    updateDatetime();
+  }, 3000);
```
Comment (JasonPuglisi) [inline public/javascripts/now.js]: Might as well change the timeout to 1000. It shouldn't make a difference performance wise, and we'll have a slightly more accurate time. Also consider using [`setInterval`](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval) (I know the existing code isn't the prettiest haha).

File: public/javascripts/now.js [hunk 4]
```diff
@@ -670,6 +672,46 @@ function updateWeather(coords, iconMap) {
...
+    'December'
+  ];
+  var date = new Date();
+  var month = months[date.getMonth()];
+  return month + ' ' + date.getDate() + ', ' + date.getFullYear();
+}
+
+function getCurrentTime() {
+  var date = new Date();
+  return date.getHours() + ':' + date.getMinutes();
```
Comment (JasonPuglisi) [inline public/javascripts/now.js]: Pad the hours and minutes left with a zero for single digits.

Optional: This is a 24hr time. Would you consider adding a 12hr option, with configuration in `/app/config`? I'd like there to be a `Time Display` section at the top above `Weather Units` with options `12-hour` and `24-hour`. The 12hr option would display `AM` or `PM` as appropriate, without a leading zero for the hour, but the 24hr option would display a leading zero for a single digit hour. I'm willing to merge without this, but just give it a thought!

File: public/stylesheets/style.css [hunk 5]
```diff
@@ -173,11 +176,15 @@ input[type=text] {
   color: #d7d7d7;
 }
 
-#music #title, #weather #summary {
+#datetime #time {
```
Comment (JasonPuglisi) [inline public/stylesheets/style.css]: Could you leave this with the default size for `h2`?

File: public/stylesheets/style.css [hunk 6]
```diff
@@ -173,11 +176,15 @@ input[type=text] {
...
+  font-size: 4em;
+}
+
+#music #title, #weather #summary, #datetime #date, #datetime #time {
```
Comment (JasonPuglisi) [inline public/stylesheets/style.css]: Add `#datetime #date` to this one instead of `#datetime #time`.

File: public/stylesheets/style.css [hunk 7]
```diff
@@ -173,11 +176,15 @@ input[type=text] {
...
   margin: 0;
 }
 
-#music #artist, #weather #conditions  {
+#music #artist, #weather #conditions {
```
Comment (JasonPuglisi) [inline public/stylesheets/style.css]: Add `#datetime #time` to this one.

**Comment by JasonPuglisi** [comment]
Looks good to me! Anything left to add or is this ready to merge?

**Comment by sprkweb** [comment]
Well, I can do the 12hr option, but later, it is too late now :)
P. S. Do you use the 12hr system in your country? It is interesting

**Comment by JasonPuglisi** [comment]
In that case, I'll merge this and open a separate issue for a 12hr option. If you feel like doing it at some point, that's great! If not, no worries.

For most people here in the United States, 12hr is used almost exclusively. I personally prefer 24hr, but to each their own :)

**Comment by JasonPuglisi** [comment]
Squashed and merged. If you plan to continue development, you'll want to reset master. Thanks a lot!

**Pull request merged**